module/apmstreadwayamqp/utils.go

```diff
@@ -0,0 +1,64 @@
+package apmstreadwayamqp
+
+import (
+	"fmt"
+	"go.elastic.co/apm/module/apmhttp/v2"
+	"go.elastic.co/apm/v2"
+	"strings"
+)
+
+var (
+	elasticTraceparentHeader = strings.ToLower(apmhttp.ElasticTraceparentHeader)
+	w3cTraceparentHeader     = strings.ToLower(apmhttp.W3CTraceparentHeader)
+	tracestateHeader         = strings.ToLower(apmhttp.TracestateHeader)
+	apmStreadwayAmqpPrefix   = "apmstreadwayamqp_transaction_"
+)
+
+func getMessageTraceparent(headers map[string]interface{}, header string) (apm.TraceContext, bool) {
+	headerValue, ok := headers[fmt.Sprintf("%s_%s", apmStreadwayAmqpPrefix, header)]
+	if !ok || headerValue == nil {
+		return apm.TraceContext{}, false
+	}
```

Comment:
```suggestion
	headerValue, ok := headers[fmt.Sprintf("%s_%s", apmStreadwayAmqpPrefix, header)].(string)
	if !ok {
		return apm.TraceContext{}, false
	}
```

Then you can remove the type switch below.